beamer/agent/events.py

```diff
@@ -315,7 +315,7 @@ def fetch(self) -> list[Event]:
                 break
             if events is not None:
                 result.extend(events)
-                from_block = BlockNumber(to_block + 1)
+            from_block = BlockNumber(to_block + 1)
 
         self._next_block_number = from_block
         try:
```

Comment:
The commit message should be improved to explain why we are doing this change, e.g. ` agent: EventFetcher: make progress even in case of no events in the block range`.